src/main/java/com/zendesk/maxwell/producer/MaxwellSNSProducer.java

```diff
@@ -0,0 +1,112 @@
+package com.zendesk.maxwell.producer;
+
+import com.amazonaws.handlers.AsyncHandler;
+import com.amazonaws.services.sns.AmazonSNSAsync;
+import com.amazonaws.services.sns.AmazonSNSAsyncClientBuilder;
+import com.amazonaws.services.sns.model.MessageAttributeValue;
+import com.amazonaws.services.sns.model.PublishRequest;
+import com.amazonaws.services.sns.model.PublishResult;
+import com.zendesk.maxwell.MaxwellContext;
+import com.zendesk.maxwell.replication.Position;
+import com.zendesk.maxwell.row.RowMap;
+import org.apache.commons.lang3.ArrayUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class MaxwellSNSProducer extends AbstractAsyncProducer {
+
+	private AmazonSNSAsync client;
+	private String topic;
+	private String[] stringFelds = {"database", "table"};
+	private String[] numberFields = {"ts", "xid"};
+
+	public MaxwellSNSProducer(MaxwellContext context, String topic) {
+		super(context);
+		this.topic = topic;
+		System.out.println(topic);
+		this.client = AmazonSNSAsyncClientBuilder.defaultClient();
+	}
+
+	public void setClient(AmazonSNSAsync client) {
+		this.client = client;
+	}
+
+	@Override
+	public void sendAsync(RowMap r, CallbackCompleter cc) throws Exception {
+		String value = r.toJSON();
+		// Publish a message to an Amazon SNS topic.
+		final PublishRequest publishRequest = new PublishRequest(topic, value);
+		Map<String, MessageAttributeValue> messageAttributes = new HashMap<>();
+
+		final String configuredAttributes = context.getConfig().snsAttrs;
+		if (configuredAttributes != null) {
+			for (String element: configuredAttributes.split(",")) {
+				switch (element) {
+					case "database":
+						messageAttributes.put(
+							"database",
+							new MessageAttributeValue().withDataType("String").withStringValue(r.getDatabase())
+						);
+						break;
+					case "table":
+						messageAttributes.put(
+							"table",
+							new MessageAttributeValue().withDataType("String").withStringValue(r.getTable())
+						);
+						break;
+				}
+			}
+		}
```

Comment:
i think this is probably fine but some purist-engineer part of me recoils and says you should parse the string once in the initializer and then compile down to enums and oh god we'll never get it merged

Reply:
I thought that as well. I opted to keep most of the logic inside of the SNS producer for now, as that's what mostly would be in charge of it. If we started allowing specific `data` fields to be mapped for example, classes upstream would need to know how to handle something like 
```
sns_attrs=database,table,sometable.somefield:String,anothertable.otherfield:Boolean` (for type casting the attributes)
```
or whatever, depending on what those interfaces actually look like when they get implemented. Or alternatively, some logic would live in the config, some would still live in the producer. So I'm not totally sure what the best strategy would be

The current usage isn't something I'd brag about writing, but I agree, its probably fine.